templates/model/spec.go

```diff
@@ -112,7 +112,7 @@ func (s *Spec) Validate() error {
 	return errors.Join(
 		oneOf(s.Pos, s.APIVersion, []string{"cli.abcxyz.dev/v1alpha1"}, "apiVersion"),
 		oneOf(s.Pos, s.Kind, []string{"Template"}, "kind"),
-		notZero(s.Pos, s.Desc, "desc"),
+		notZeroModel(s.Pos, s.Desc, "desc"),
```

Comment:
Note for reviewers: I renamed this because now I want two versions of this function: one that takes a value wrapped in a valWIthPos[T], and another one that just takes a T.